RetrievingAndDeleting.md

```diff
@@ -30,28 +33,52 @@ The following code can be added to the FoodTracker app and will query the saved
+4. Call this new function by replacing the `loadMeals` function with:
+```swift
+private func loadMeals() -> [Meal]?  {
+        loadFromServer()
+        return meals
+    }
+```
+5. Inside your `saveMeals` function, replace:
+```swift
+do {
+    let data = try PropertyListEncoder().encode(meals)
+    let isSuccessfulSave  = NSKeyedArchiver.archiveRootObject(data, toFile: MealTableViewController.ArchiveURL.path)
+    if isSuccessfulSave {
+        os_log("Meals successfully saved.", log: OSLog.default, type: .debug)
+    } else {
+        os_log("Failed to save meals...", log: OSLog.default, type: .error)
+    }
+} catch {
+    os_log("Failed to save meals...", log: OSLog.default, type: .error)
+}
+```  
+with:
+```swift
+loadFromServer()
+```
 
+This change makes the food tracker app to load it's meals from the server's database, instead of a local cache. You can test this by closing then deleting the app in the simulator and restarting the food tracker. All your meals from the Server will persist since they have been loaded from the database.
+### Deleting Meals from the FoodServer  
 Support for deleting Meals from the FoodServer can be added by adding a REST API that responds to a DELETE request, with an additional identifier that provides information about which meal to delete.
-
-1. The following code adds a DELETE handler to the FoodServer for a single Meal:  
+1. Open the FoodServer project
+`open  ~/FoodTrackerBackend/FoodServer/FoodServer.xcodeproj/`
+2. Add the following code as a DELETE handler for a single Meal:
 ```swift
-    router.delete("/meal", handler: deleteHandler)
+router.delete("/meal", handler: deleteHandler)
 ```  
-With the following handler implementation:  
+With the following deleteHandler implementation:  
 ```swift
-    func deleteHandler(id: String, completion: (RequestError?) -> Void ) {
-        print("Deleting \(id) from mealStore")
-        mealStore[id] = nil
-        completion(nil)
-    }
+func deleteHandler(id: String, completion: @escaping (RequestError?) -> Void ) {
+    Meal.delete(id: id, completion)
+}
 ```
-Note that rather than receiving a Meal, this receives an `id`. The `id` is an `Identifier` that is added to the URL in order to denote which Meal to delete. Kitura extends String to conform to Identifier, so strings can be used directly.  
+Note that rather than receiving a Meal, this receives an `id`. The `id` is an `Identifier` that is added to the URL in order to denote which Meal to delete. Kitura extends String to conform to Identifier, so strings can be used directly. Here, we will be using the meal name as the `Identifier`.
 
-2. The following code can be used in the FoodTracker app to make the call to the FoodServer. Here the Meal name is used for the Identifier.  
+3. The following function is added to the FoodTracker > MealTableViewController to make the delete call to the FoodServer.
```

Comment:
I would just specify the filename here, like `MealTableViewController.swift` instead of the hierarchy you used here.